apps/hash-frontend/src/pages/[shortname]/entities/[entity-uuid].page/entity-editor/file-preview-section.tsx

```diff
@@ -1,50 +1,237 @@
+            </GrayToBlueIconButton>
+          </Box>
+        </Tooltip>
+      </ActionButtonsContainer>
+    </>
+  );
+};
+
+export const FilePreviewSection = () => {
+  const [replacing, setReplacing] = useState(false);
+
+  const { isDirty, entitySubgraph } = useEntityEditor();
+
+  const entity = getRoots(entitySubgraph)[0]!;
 
-  const title = generateEntityLabel(entitySubgraph);
+  const { isImage, url } = getFileUrlFromFileProperties(entity.properties);
+
+  const { description, displayName, fileName } = simplifyProperties(
+    entity.properties as FileProperties,
+  );
+
+  const title = displayName ?? generateEntityLabel(entitySubgraph);
 
   const alt = description ?? title;
 
   return (
     <SectionWrapper title="File Preview">
-      <Box
+      <Stack
         sx={({ boxShadows }) => ({
+          alignItems: "center",
+          justifyContent: "center",
           boxShadow: boxShadows.sm,
           borderRadius: 1,
-          maxHeight: maxImageHeight,
+          height: previewHeight,
+          position: "relative",
         })}
       >
-        <ImageWithCheckedBackground
-          alt={alt}
-          src={imageUrl}
-          sx={{ maxHeight: maxImageHeight }}
-        />
-      </Box>
+        {replacing ? (
+          <ReplaceFile
+            close={() => setReplacing(false)}
+            description={description}
+            displayName={displayName}
+            isImage={!!isImage}
+          />
+        ) : (
+          <>
+            <ActionButtonsContainer>
+              <Tooltip
+                title={
+                  isDirty
+                    ? "Save or discard your changes to replace the file"
+                    : "Replace"
+                }
+              >
+                <Box>
+                  <GrayToBlueIconButton
+                    disabled={isDirty}
+                    onClick={() => setReplacing(true)}
+                  >
+                    <RotateIcon sx={{ width: 13, height: 13 }} />
+                  </GrayToBlueIconButton>
+                </Box>
+              </Tooltip>
+              <Tooltip title="Download">
+                <Box
+                  component="a"
+                  href={url}
+                  target="_blank"
+                  rel="nofollow noopener noreferrer"
+                >
+                  <GrayToBlueIconButton>
+                    <DownloadIcon sx={{ width: 13, height: 13 }} />
+                  </GrayToBlueIconButton>
+                </Box>
+              </Tooltip>
+            </ActionButtonsContainer>
+            {isImage ? (
+              <ImageWithCheckedBackground
+                alt={alt}
+                src={url}
+                sx={{ height: previewHeight }}
+              />
+            ) : (
+              <Stack alignItems="center" spacing={2}>
+                <FileIcon sx={{ color: "gray.50", fontSize: 48 }} />
+                <Typography sx={{ color: "gray.70" }}>{fileName}</Typography>
```

Comment:
```suggestion
                <FileIcon
                  sx={{
                    color: ({ palette }) => palette.gray[50],
                    fontSize: 48,
                  }}
                />
                <Typography sx={{ color: ({ palette }) => palette.gray[70] }}>
                  {fileName}
                </Typography>
```

Reply:
Is there any functional difference to this, or is it for consistency?

Reply:
It [seems not](https://mui.com/system/getting-started/the-sx-prop/#theme-aware-properties), but happy to use this style if we'd rather

Reply:
I mainly do it to get type safety when accessing the colors, which prevents the possiblity for typos being un-caught and incase we ever change the palette again (hopefully we don't) we can find where we are accessing the values more easily.

Reply:
That's a good reason to, I have updated in https://github.com/hashintel/hash/pull/3216/commits/a71ba3dabbb543c7db831db03b46bf4e3619904a